### ISSUE TITLE
why umap showing bbknn perfectively integrated but tsne 

### ISSUE THREAD

**Issue by honghh2018** [scverse/scanpy#1370]
Hi @all,
    Thanks to develop the great tools,
     I encounter a pecular problem on bbknn integrated data.
     i follow the workflow code to run the data integrated,the code showing below,
#
     sc.pp.highly_variable_genes(adata)
adata = adata[:, adata.var['highly_variable']]
sc.tl.pca(adata, svd_solver='arpack')
sc.tl.tsne(adata)
///data integrated
sc.external.pp.bbknn(adata, batch_key='orig.ident')

sc.tl.umap(adata)
adata
sc.pl.umap(adata, color=['orig.ident'])
showing the well integrated, picture below,
![image](https://user-images.githubusercontent.com/41668708/90249921-df74d980-de6d-11ea-8283-3830dd3c6ad2.png)
But,when i want to see the tsne picture, the batch from different sample showing up on the tsne but umap like above picture
,i runing ,
sc.tl.tsne(adata)
sc.pl.tsne(adata, color=['orig.ident'])
the picture show below, indicating that the integrated can not be worked on tsne.
![image](https://user-images.githubusercontent.com/41668708/90250154-3c708f80-de6e-11ea-8d43-7a597c5693a2.png)
So, why this tsne showing significantly different with the object just running over the integrated process.
any advice would be appreciated
Best,
hanhuihong

**Comment by hejian41** [comment]
same issue, any suggestions?

**Comment by LuckyMD** [comment]
Hi @honghh2018 and @hejian41,

This question might be better addressed to the developers of `BBKNN` here: https://github.com/Teichlab/bbknn

**Comment by ivirshup** [comment]
I think the issue here is that BBKNN only generates an integrated graph, while the tsne computation creates a new graph from some matrix representation of the data. There has been the suggestion of allowing a tsne layout (#1233) to be generated from a precomputed connectivity matrix, but that hasn't been implemented here yet.

**Comment by LuckyMD** [comment]
Ah, I wasn't aware that the `sc.tl.tsne()` function doesn't use the `sc.pp.neighbors()` output. Although I do recall the issue about this... Thanks for the context!

**Comment by atarashansky** [comment]
One potential solution is to convert the integrated connectivity matrix, C, into a pseudo-distance matrix (1-C) (this probably won't work for datasets much larger than 10k cells due to memory limitations) and run t-SNE with the 'precomputed' metric on that fake distance matrix. If scanpy's t-SNE wrapper does not allow passing a precomputed distance matrix, I would recommend using the sklearn implementation directly:

https://scikit-learn.org/stable/modules/generated/sklearn.manifold.TSNE.html
